source/material_model/viscoelastic_plastic.cc

```diff
@@ -0,0 +1,737 @@
+                             "the model to run the user must select 'true' or 'false'.");
+          prm.declare_entry ("Fixed elastic time step", "1.e3",
+                             Patterns::Double (0),
+                             "The fixed elastic time step $dte$. Units: years if the "
+                             "'Use years in output instead of seconds' parameter is set; "
+                             "seconds otherwise.");
+          prm.declare_entry ("Use stress averaging","false",
+                             Patterns::Bool (),
+                             "Whether to apply a stress averaging scheme to account for differences "
+                             "between the fixed elastic time step and numerical time step. ");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+    ViscoelasticPlastic<dim>::parse_parameters (ParameterHandler &prm)
+    {
+
+      // Get the number of fields for composition-dependent material properties
+      const unsigned int n_fields = this->n_compositional_fields() + 1;
+
+      prm.enter_subsection("Material model");
+      {
+        prm.enter_subsection("Viscoelastic Plastic");
+        {
+
+          AssertThrow(this->get_parameters().enable_elasticity == true,
+                      ExcMessage ("Material model Viscoelastic only works if 'Enable elasticity' is set to true"));
+
+          reference_temperature = prm.get_double ("Reference temperature");
+          minimum_strain_rate = prm.get_double("Minimum strain rate");
+          reference_strain_rate = prm.get_double("Reference strain rate");
+          minimum_viscosity = prm.get_double ("Minimum viscosity");
+          maximum_viscosity = prm.get_double ("Maximum viscosity");
+          input_reference_viscosity = prm.get_double ("Reference viscosity");
+
+          viscosity_averaging = MaterialUtilities::parse_compositional_averaging_operation ("Viscosity averaging scheme",
+                                prm);
+
+
+          // Plasticity parameters
+          angles_internal_friction = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Angles of internal friction"))),
+                                                                             n_fields,
+                                                                             "Angles of internal friction");
+          // Convert angles from degrees to radians
+          for (unsigned int i = 0; i<n_fields; ++i)
+            angles_internal_friction[i] *= numbers::PI/180.0;
+          cohesions = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Cohesions"))),
+                                                              n_fields,
+                                                              "Cohesions");
+
+          // Parse viscoelastic properties
```

Comment:
this comment is copy-pasted from `ViscoElastic` and is confusing here. Either rename to `ViscoElasticPlastic properties` or move down to the elasticity parameters.

Reply:
Agree this is a bit confusing. I reworded the comment to `Parse additional material properties`.